django/contrib/admin/utils.py

```diff
@@ -340,12 +340,9 @@ def label_for_field(name, model, model_admin=None, return_attr=False):
             # field is likely a ForeignObjectRel
             label = field.related_model._meta.verbose_name
     except FieldDoesNotExist:
-        if name == "__unicode__":
+        if name == "__str__":
```

Comment:
I agree with your analysis.  These lines are from 2304ca423640a7b06390530bf61c879f9ff9e4ba and the test was later removed in 65cc646c4801e3f3e1df1ab06dfaa763fa4b7b22. Opening a new ticket for this issue is fine.